### PR TITLE
[Enhancement](sequence-column) optimize the use of sequence column

### PR DESCRIPTION
**Pull request by liaoxin01**
# Proposed changes

Issue Number: close #xxx

## Problem summary

When you create the Uniq table, you can specify the mapping of sequence column to other columns.  You no longer need to specify mapping column when importing.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 38.5 seconds
 load time: 610 seconds
 storage size: 17154865268 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221101153320_clickbench_pr_37657.html

File: fe/fe-core/src/main/java/org/apache/doris/planner/BrokerScanNode.java [hunk 1]
```diff
@@ -277,9 +278,13 @@ private void initColumns(ParamCreateContext context) throws UserException {
                 columnDescs.descs.add(ImportColumnDesc.newDeleteSignImportColumnDesc(new IntLiteral(1)));
             }
             // add columnExpr for sequence column
-            if (context.fileGroup.hasSequenceCol()) {
+            if (targetTable instanceof OlapTable && ((OlapTable) targetTable).hasSequenceCol()) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/BrokerScanNode.java:281]: Please also modify `LoadScanProvider`, line 194

File: fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java [hunk 2]
```diff
@@ -893,6 +893,21 @@ public void setBloomFilterInfo(Set<String> bfColumns, double bfFpp) {
...
+        }
+        return tableProperty.getSequenceMapCol();
+    }
+
+    // map the sequence column to other column
+    public void setSequenceMapCol(String colName) {
+        if (tableProperty == null) {
+            tableProperty = new TableProperty(new HashMap<>());
+        }
+        tableProperty.setSequenceMapCol(colName);
```
[the remark this answers is not in the document]
Comment (liaoxin01) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java:908]: `this.hasSequenceCol` is set by `setSequenceInfo` function in InternalCatalog line 2026.

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java [hunk 3]
```diff
@@ -4125,6 +4131,11 @@ private void renameColumn(Database db, OlapTable table, String colName,
             }
         }
 
+        // 5. modify sequence map col
+        if (table.hasSequenceCol() && table.getSequenceMapCol().equalsIgnoreCase(colName)) {
```
[the remark this answers is not in the document]
Comment (liaoxin01) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java:4135]: done.

File: fe/fe-core/src/main/java/org/apache/doris/planner/BrokerScanNode.java (hunk 1 above)
Comment (liaoxin01) [reply fe/fe-core/src/main/java/org/apache/doris/planner/BrokerScanNode.java:281]: done.

File: docs/en/docs/data-operate/update-delete/sequence-column-manual.md [hunk 4]
```diff
@@ -60,13 +60,21 @@ The principle is the same as the reading process during Base Compaction.
...
+1. When you create the Uniq table, you can specify the sequence column type.
 ```text
 PROPERTIES (
     "function_column.sequence_type" = 'Date',
 );
 ```
-The sequence_type is used to specify the type of the sequence column, which can be integral and time (DATE / DATETIME).
+The sequence_type is used to specify the type of the sequence column, which can be integral and time (DATE / DATETIME). The mapping column needs to be specified when importing.
+
+2. When you create the Uniq table, you can specify the mapping of sequence column to other columns.
```
[the remark this answers is not in the document]
Comment (liaoxin01) [reply docs/en/docs/data-operate/update-delete/sequence-column-manual.md]: done.

File: docs/zh-CN/docs/data-operate/update-delete/sequence-column-manual.md [hunk 5]
```diff
@@ -64,15 +64,24 @@ Sequence列建表时在property中增加了一个属性，用来标识`__DORIS_S
...
+方法一：创建Uniq表时，可以指定sequence列到其他column的映射。
+
+```text
+PROPERTIES (
+    "function_column.sequence_col" = 'column_name',
+);
+```
+sequence_col用来指定sequence列到其他column的映射，该列可以为整型和时间类型（DATE、DATETIME），创建后不能更改该列的类型。导入时不用再指定映射列。如果设置了`function_column.sequence_col`, `function_column.sequence_type`将被忽略。
+
+方法二：创建Uniq表时，可以指定sequence列类型。
```
Comment (zhannngchen) [inline docs/zh-CN/docs/data-operate/update-delete/sequence-column-manual.md:76]: We'd better hide the usage of method 2, it's only for compatibility to keep the old impl.

File: fe/fe-core/src/test/java/org/apache/doris/catalog/CreateTableTest.java [hunk 6]
```diff
@@ -208,6 +208,15 @@ public void testNormal() throws DdlException {
                         + "distributed by hash(k2) buckets 1\n"
                         + "properties('replication_num' = '1');"));
 
+        // table with sequence col
+        ExceptionChecker
```
Comment (zhannngchen) [inline fe/fe-core/src/test/java/org/apache/doris/catalog/CreateTableTest.java:212]: We also need a ut for alter table?

File: docs/zh-CN/docs/data-operate/update-delete/sequence-column-manual.md (hunk 5 above)
Comment (zhannngchen) [reply docs/zh-CN/docs/data-operate/update-delete/sequence-column-manual.md:76]: 后边的例子都得跟着改一下，去掉那些 load 的用法说明，最后的示例也要跟着改掉

File: fe/fe-core/src/test/java/org/apache/doris/catalog/CreateTableTest.java (hunk 6 above)
Comment (liaoxin01) [reply fe/fe-core/src/test/java/org/apache/doris/catalog/CreateTableTest.java:212]: done.

File: docs/zh-CN/docs/data-operate/update-delete/sequence-column-manual.md (hunk 5 above)
Comment (liaoxin01) [reply docs/zh-CN/docs/data-operate/update-delete/sequence-column-manual.md:76]: 感觉可以先留着，原来的方式比较灵活，有些场景用户可以就要单独指定一列数据作为sequence列，先试用看看，逐步再去删掉原来的方案。

File: docs/en/docs/data-operate/update-delete/sequence-column-manual.md [hunk 7]
```diff
@@ -60,13 +60,21 @@ The principle is the same as the reading process during Base Compaction.
 When the Sequence column creates a table, an attribute is added to the property, which is used to identify the type import of `__DORIS_SEQUENCE_COL__`. The grammar design is mainly to add a mapping from the sequence column to other columns. The settings of each seed method will be described below introduce.
 
 #### Create Table
-When you create the Uniq table, you can specify the sequence column type.
+1. When you create the Uniq table, you can specify the mapping of sequence column to other columns.
+```text
+PROPERTIES (
+    "function_column.sequence_col" = 'column_name',
```
Comment (zhannngchen) [inline docs/en/docs/data-operate/update-delete/sequence-column-manual.md:66]: pls also update `create-table.md`

Comment (liaoxin01) [reply docs/en/docs/data-operate/update-delete/sequence-column-manual.md:66]: done.

File: fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java [hunk 8]
```diff
@@ -574,6 +574,12 @@ private void processModifyColumn(ModifyColumnClause alterClause, OlapTable olapT
             throw new DdlException("Column[" + columnPos.getLastCol() + "] does not exists");
         }
 
+        // sequence col can not change type
```
[the remark this answers is not in the document]
Comment (liaoxin01) [reply fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java:577]: It is not yet supported to add the property by alter table.

**Comment by zhannngchen** [comment]
LGTM

**Review by zhannngchen — APPROVED** [review]

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**